test/envtest/scheme.go

```diff
@@ -0,0 +1,43 @@
+package envtest
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	kongv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+	kongv1alpha1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1alpha1"
+	kongv1beta1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1beta1"
+)
+
+type SchemeOption func(t *testing.T, s *k8sruntime.Scheme)
+
+// WithGatewayAPI registers the Gateway API types with the scheme.
+func WithGatewayAPI(t *testing.T, s *k8sruntime.Scheme) {
+	require.NoError(t, gatewayv1beta1.AddToScheme(s))
+	require.NoError(t, gatewayv1alpha2.AddToScheme(s))
+}
+
+// WithKong registers the Kong types with the scheme.
+func WithKong(t *testing.T, s *k8sruntime.Scheme) {
```

Comment:
nit: I'd expect to see those to return a func so that the call site looks like this

```
	scheme := Scheme(t, WithGatewayAPI())
```

I wonder if there's a significant difference and/or drawback of using pure function pointers 🤔

Reply:
Fair enough. Let's leave it as is for now and extend if need be ( when we need a param for those option funcs)